adbe/adb_enhanced.py

```diff
@@ -1504,6 +1504,14 @@ def perform_uninstall(app_name, first_user):
         # https://www.xda-developers.com/uninstall-carrier-oem-bloatware-without-root-access/
         cmd = "--user 0"
     return_code, _, stderr = execute_adb_shell_command2('pm uninstall %s %s' % (cmd, app_name))
+    if return_code == 0:
+        return
+
+    if not cmd:
+        print("Uninstall failed, retry with first user")
```

Comment:
How about`print_msg("Uninstall failed, trying to uninstall for user 0...")`?

Reply:
While testing the uninstall with multiple devices without specifying which one to use, the error message is:
```
adb: more than one device/emulator

Package com.android.package does not exist
```
Is it wanted ?

Reply:
> adb: more than one device/emulator

Yeah, we don't support more than one device simultaneously. It complicates everything, what if there is a command that succeeds on one device and fails on the other? What if there are 10 devices? And what about commands like `apps list` that have a long output, how to make that readable/parseable for more than one device?

That's why I feel it is best is to support only one device at a time and let the caller specify that device ID with `-s`.

Reply:
Got it, if no device is selected, I think that only the error message `adb: more than one device/emulator`should be displayed then the program exit without displaying the second error message `Package com.android.package does not exist` as it can exist in one of the connected devices.

Reply:
I agree.